packages/core/src/components/ModalNew/Modal/Modal.tsx

```diff
@@ -43,27 +42,57 @@ const Modal = forwardRef(
       [id, setTitleIdCallback, setDescriptionIdCallback]
     );
 
+    const onBackdropClick = useCallback<React.MouseEventHandler<HTMLDivElement>>(
+      e => {
+        if (!show) return;
+        onClose(e);
+      },
+      [onClose, show]
+    );
+
+    const onEscClick = useCallback<React.KeyboardEventHandler<HTMLBodyElement>>(
+      e => {
+        if (!show) return;
+        onClose(e);
+      },
+      [onClose, show]
+    );
+
+    useKeyEvent({
+      callback: onEscClick,
+      capture: true,
+      keys: ["Escape"]
```

Comment:
I think we have consts for these

Reply:
Will change for now, but I think we should make `useKeyEvent` use string keys as well, same as we switched from static props to inline string props

can be a non-breaking change after we release Vibe3

Reply:
But it's not type-safe, no?

Reply:
it can be
we can extract the keys strings from React's TS I believe